### PR TITLE
Adds OSS::Attribute::RevealableEmail + tests + docs

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?
Adds OSS::Attribute::RevealableEmail + tests + docs

https://github.com/upfluence/oss-components/assets/5032005/0b9dd6d6-ab7f-4ebb-937d-bf76adf31807

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/attribute/revealable-email.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { assert } from '@ember/debug';
+
+interface OSSAttributeRevealableEmailArgs {
+  tooltip?: string;
+  lockTooltip?: string;
+  onRevealEmail(): Promise<void>;
+}
+
+export default class OSSAttributeRevealableEmail extends Component<OSSAttributeRevealableEmailArgs> {
+  @service intl: any;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/attribute/revealable-email.ts:14]:
```suggestion
  @service declare intl: any;
```

File: tests/dummy/app/controllers/application.js [hunk 2]
```diff
@@ -137,6 +138,28 @@ export default class ApplicationController extends Controller {
...
+      setTimeout(() => {
+        this.toast.error('There was an error retrieving the email address. Try again later.', 'Error');
+        return rej('failed');
+      }, 1000);
+    });
+  }
+
+  @action
+  onRevealEmailSuccess() {
+    console.log('on reveal email error');
```
Comment (JulienVannier66) [inline tests/dummy/app/controllers/application.js:154]:
```suggestion
  onRevealEmailSuccess() {
    console.log('on reveal email success');
```

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: addon/components/o-s-s/attribute/revealable-email.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+
+interface OSSAttributeRevealableEmailArgs {
+  tooltip?: string;
+  lockTooltip?: string;
+  onRevealEmail(): Promise<void>;
```
Comment (phndiaye) [inline addon/components/o-s-s/attribute/revealable-email.ts:10]: nit: what about `Promise<unknown>` as we don't really know what the parent

**Review by phndiaye — APPROVED** [review]

**Pull request merged**